src/jit/morph.cpp

```diff
@@ -10637,8 +10654,17 @@ GenTree* Compiler::fgMorphRecognizeBoxNullable(GenTree* compare)
         return compare;
     }
 
-    // replace the box with an access of the nullable 'hasValue' field which is at the zero offset
-    GenTree* newOp = gtNewOperNode(GT_IND, TYP_BOOL, opCall->gtCall.gtCallArgs->gtOp.gtOp2->gtOp.gtOp1);
+    // Get the nullable struct argument
+    GenTreePtr arg = opCall->gtCall.gtCallArgs->gtOp.gtOp2->gtOp.gtOp1;
```

Comment:
I'd use `GenTree*`, if only because couple of lines below we also have `GenTree*`. In general, this `GenTreePtr` thing should IMO be avoided. It's pointless and inconsistent.